src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/UseLiteralsWhereAppropriate.Fixer.cs

```diff
@@ -20,11 +26,84 @@ public sealed override FixAllProvider GetFixAllProvider()
             return WellKnownFixAllProviders.BatchFixer;
         }
 
-        public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
-            // Fixer not yet implemented.
-            return Task.CompletedTask;
-            
+            SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            SyntaxNode declaration = root.FindNode(context.Span);
+            declaration = SyntaxGenerator.GetGenerator(context.Document).GetDeclaration(declaration, DeclarationKind.Field);
+            var fieldFeclaration = GetFieldDeclaration(declaration);
+            if (fieldFeclaration == null)
+            {
+                return;
+            }
+
+            // We cannot have multiple overlapping diagnostics of this id.
+            Diagnostic diagnostic = context.Diagnostics.Single();
+
+            context.RegisterCodeFix(
+                new MyCodeAction(
+                    MicrosoftQualityGuidelinesAnalyzersResources.UseLiteralsWhereAppropriateCodeActionTitle,
+                    cancellationToken => ToConstantDeclarationAsync(context.Document, fieldFeclaration, cancellationToken)),
+                diagnostic);
+        }
+
+        private async Task<Document> ToConstantDeclarationAsync(Document document, SyntaxNode fieldDeclaration, CancellationToken cancellationToken)
+        {
+            var editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+
+            SyntaxTriviaList leadingTrivia = new SyntaxTriviaList();
+            SyntaxTriviaList trailingTrivia = new SyntaxTriviaList();
+
+            SyntaxTokenList newModifiers = new SyntaxTokenList();
+            foreach (SyntaxToken modifier in GetModifiers(fieldDeclaration))
+            {
+                if (IsStaticKeyword(modifier) || IsReadonlyKeyword(modifier))
+                {
+                    if (leadingTrivia.Count == 0 && trailingTrivia.Count == 0)
+                    {
+                        leadingTrivia = leadingTrivia.AddRange(modifier.LeadingTrivia);
+                        trailingTrivia = trailingTrivia.AddRange(modifier.TrailingTrivia);
+                    }
+                    else
+                    {
+                        trailingTrivia = trailingTrivia.AddRange(modifier.LeadingTrivia);
```

Comment:
Actually this is intentional. The associated analyzer ensures we'll only get in the fixer if both `static` and `readonly` keywords are in the declaration. Because their order is not relevant, we detect if both have been passed  by inspecting whether leading and trailing trivia are empty (the `if` condition). This `else` case copies the trivia in-between both keywords into the combined set of trailing trivia. 

For example:

```csharp
public /*1*/ readonly /*2*/ static /*3*/ string f;
```

is changed into:

```csharp
public /*1*/ const /*2*/ /*3*/ string f;
```

Here you can see that `2` moved from leading trivia to trailing trivia of the `const` keyword.